third-party/jemalloc/README

```diff
@@ -0,0 +1,11 @@
+==========================
+jemalloc README for Chapel
+==========================
+
+This copy of jemalloc is being released with Chapel for convenience and
+uses a snapshot from:
```

Comment:
Yeah, I guess I should be linking to the github repo, instead of the landing page: https://github.com/jemalloc/jemalloc like we do with tcmalloc